napari_plugin_devtools/validation.py

```diff
@@ -0,0 +1,126 @@
+import inspect
+import json
+import os
+from types import FunctionType
+from typing import Callable, Dict, List, Tuple, Union
+
+from napari.plugins import plugin_manager
+from napari_plugin_engine import HookImplementation
+from pkginfo import SDist, Wheel
+
+functions: Dict[Tuple[str, str], Callable] = dict()
+
+
+def validate_artifacts(folder):
+    """Validate the artifacts built for distribution.
+
+    Parameters
+    ----------
+    folder : str
+        artifacts folder after build with setuptools
+
+    Raises
+    ------
+    AssertionError
+        If the artifact is not build properly,
+    """
+    for pkgpath in os.listdir(folder):
+        pkgpath = os.path.join(folder, pkgpath)
+        if pkgpath.endswith('.tar.gz'):
+            dist = SDist(pkgpath)
+        elif pkgpath.endswith('.whl'):
+            dist = Wheel(pkgpath)
+        else:
+            print(f'Not a valid format {pkgpath}')
```

Comment:
Longer term we'll want to think about what sort of notifications to give - do we raise an error, a warning, something else etc.

Fine to merge now with prints as we get going, but something to think about longer term

Reply:
yeah, definitely finalizing this when I am running it against real plugins